tests/queries/0_stateless/02947_limit_materialized_view_size.sql

```diff
@@ -0,0 +1,32 @@
+DROP TABLE IF EXISTS 02947_tb;
+DROP TABLE IF EXISTS 02947_tb_view1;
+DROP TABLE IF EXISTS 02947_tb_view2;
+
+CREATE TABLE 02947_tb
+(
+    a String,
+    b String,
+    c String,
+)
+ENGINE = MergeTree
+ORDER BY a;
+
+SET max_materialized_views_account_for_table = 1;
+
+CREATE MATERIALIZED VIEW 02947_tb_view1
+ENGINE = MergeTree
+ORDER BY b
+AS SELECT * FROM 02947_tb;
+
+CREATE MATERIALIZED VIEW 02947_tb_view2
+ENGINE = MergeTree
+ORDER BY c
+AS SELECT * FROM 02947_tb;  -- { serverError TOO_MANY_MATERIALIZED_VIEWS }
+
+CREATE MATERIALIZED VIEW 02947_tb_view2
+ENGINE = MergeTree
+ORDER BY c
+AS SELECT * FROM 02947_tb
+SETTINGS max_materialized_views_size_for_table = 2;
```

Comment:
it looks a little bit strange as a query-level setting. shouldn't it be a server setting instead?

Reply:
Yes, I have changed to service settings.